buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -63,6 +66,30 @@ open class DistributionTest : Test() {
         jvmArgumentProviders.add(libsRepository)
         systemProperty("java9Home", project.findProperty("java9Home") ?: System.getProperty("java9Home"))
     }
+
+    override fun executeTests() {
+        setTimeoutMonitor()
+        super.executeTests()
+    }
+
+    private
+    fun setTimeoutMonitor() {
+        Timer(true).schedule(timerTask {
+            project.javaexec {
+                classpath = project.getConfigurations().getByName("integTestRuntimeClasspath")
```

Comment:
❌ I guess we could also use the classpath property of the current class right? (`Test.getClasspath()`).

I would actually prefer to not use `JavaExec` here but run a class in `buildSrc` mabye. WDYT?

Reply:
Changed to `Test.getClasspath()`. However, I wonder how to reuse code in `buildSrc` and `internal-integ-test` subproject. The reason I use `JavaExec` is that I don't want to copy the code to `buildSrc`.